src/main/java/com/example/demo/validation/FieldMatchesValidator.java

```diff
@@ -0,0 +1,15 @@
+package com.example.demo.validation;
+
+import com.example.demo.dto.user.UserRegistrationRequestDto;
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.util.Objects;
+
+public class FieldMatchesValidator
+        implements ConstraintValidator<FieldMatches, UserRegistrationRequestDto> {
```

Comment:
```suggestion
        implements ConstraintValidator<FieldMatches, Object> {
```
make this realization abstract and reuse in the future.